src/main/java/org/skriptlang/skript/bukkit/damagesource/elements/ExprSecDamageSource.java

```diff
@@ -0,0 +1,131 @@
+package org.skriptlang.skript.bukkit.damagesource.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.Node;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.SectionExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Trigger;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+import org.bukkit.damage.DamageSource;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.damagesource.DamageSourceExperiment;
+import org.skriptlang.skript.bukkit.damagesource.DamageSourceWrapper;
+import org.skriptlang.skript.log.runtime.SyntaxRuntimeErrorProducer;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Name("Damage Source")
+@Description({
+	"Create a custom damage source and change the attributes.",
+	"When setting a 'causing entity' you must also set a 'direct entity'.",
+	"Cannot change any attributes of a damage source outside the 'custom damage source' section."
+})
+@Example("""
+	set {_source} to a new custom damage source:
+		set the damage type to magic
+		set the causing entity to {_player}
+		set the direct entity to {_arrow}
+		set the damage location to location(0, 0, 10)
+	damage all players by 5 using {_source}
+	""")
+@Example("""
+	on damage:
+		if the damage type of event-damage source is magic:
+			set the damage to damage * 2
+	""")
+@Since("INSERT VERSION")
+@RequiredPlugins("Minecraft 1.20.4+")
+public class ExprSecDamageSource extends SectionExpression<DamageSource> implements SyntaxRuntimeErrorProducer, DamageSourceExperiment {
+
+	public static class DamageSourceSectionEvent extends Event {
+
+		private DamageSource damageSource;
+
+		public DamageSourceSectionEvent(DamageSource damageSource) {
+			this.damageSource = damageSource;
+		}
+
+		public DamageSource getDamageSource() {
+			return damageSource;
+		}
+
+		@Override
+		public @NotNull HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+	}
+
+	static {
+		Skript.registerExpression(ExprSecDamageSource.class, DamageSource.class, ExpressionType.SIMPLE,
+			"[a] [new] custom damage source");
```

Comment:
```suggestion
			"[a] custom damage source");
```
What's your thoughts on forcing a type in the expression itself? It would also mean the custom damage source class can enforce always having a type by accepting it in the constructor.

Reply:
I mean, adding it here wouldn't be all that bad. Currently I have the `DamageType` default to `GENERIC` so users could just set the `direct entity`, `causing entity` and `damage location`.

Reply:
It sounds reasonable to have it optional in the exprsec itself, and also for it to be required when constructing a damage source